odoo/addons/base/models/ir_ui_view.py

```diff
@@ -347,23 +361,24 @@ def _compute_invalid_locators(self):
                 if spec.tag == 'data':
                     specs.extend(spec)
                     continue
-                # Capture 'move' nodes to handles cases where move operations are nested within other xpath operations
-                # <xpath expr="parent" position="after">
-                #   <xpath expr="child" position="move"/>
-                # </xpath>
-                specs.extend(c for c in spec if c.get("position") == 'move')
-                node = None
-                with suppress(ValidationError):  # Syntax error
-                    # If locate_node returns None here:
-                    # Invalid expression: Ok Syntax, but cannot be anchored to the parent view.
-                    node = self.locate_node(source, spec)
-                if node is None:
-                    invalid_locators.append({
-                        "tag": spec.tag,
-                        "attrib": dict(spec.attrib),
-                        "sourceline": spec.sourceline
-                    })
+
+                invalid_locator = assess_locator(source, spec)
+                if invalid_locator is not None:
+                    invalid_locators.append(invalid_locator)
                 else:
+                    position, mode = spec.get("position"), spec.get("mode")
+                    for sub_spec in spec:
+                        if sub_spec.get("position"):
+                            if sub_spec.get("position") != "move" or (position == "replace" and mode == "inner"):
+                                invalid_locators.append({
+                                    "tag": sub_spec.tag,
+                                    "attrib": dict(sub_spec.attrib),
+                                    "sourceline": sub_spec.sourceline,
+                                })
+                            else:
+                                invalid_move = assess_locator(source, sub_spec)
+                                if invalid_move:
+                                    invalid_locators.append(invalid_move)
```

Comment:
nitpicking: walrus + invertion of the condition.
```suggestion
                        sub_position = sub_spec.get("position")
                        if sub_position == "move" and (position != "replace" or mode != "inner"):
                            if invalid_move := assess_locator(source, sub_spec):
                                invalid_locators.append(invalid_move)
                        elif sub_position:
                            invalid_locators.append({
                                "tag": sub_spec.tag,
                                "attrib": dict(sub_spec.attrib),
                                "sourceline": sub_spec.sourceline,
                            })
```